### PR TITLE
Query: Fixes Persisted continuationToken issue by turning off ODE by default

### PR DESCRIPTION
**Pull request by adityasa**
# Pull Request Template

## Description

This change fixes the issue with persisted continuation tokens by turning off ODE by Default. The issue is hit by DataVerse team where continuation tokens for a given query are persisted for long periods of time. If query that originally started with ODE resumes execution after partition split has occurred, it currently fails with following error message:
```Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: BadRequest (400); Substatus: 20007; ActivityId: ; Reason: (Response status code does not indicate success: BadRequest (400); Substatus: 20007; ActivityId: ; Reason: (Invalid format for continuation token {"OptimisticDirectExecutionToken":{"token":"{"OptimisticDirectExecutionToken". . .```

Currently this issue is tracked by https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4432, however the fix for that issue is much more involved, since it requires hydrate a continuation token required by non-ODE pipelines from original ODE continuation token.

As a mitigation, this change turns off ODE by default, unless a continuation of an ODE query is being executed.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CosmosQueryExecutionContextFactory.cs [hunk 1]
```diff
@@ -758,19 +758,14 @@ private static Documents.PartitionKeyDefinition GetPartitionKeyDefinition(InputP
             ITrace trace)
         {
             bool clientDisableOptimisticDirectExecution = await cosmosQueryContext.QueryClient.GetClientDisableOptimisticDirectExecutionAsync();
+            bool isODEContinuationToken = inputParameters.InitialUserContinuationToken != null &&
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CosmosQueryExecutionContextFactory.cs:761]: >ODE [](http://example.com/codeflow?start=19&length=3)

Ode

File: Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CosmosQueryExecutionContextFactory.cs [hunk 2]
```diff
@@ -817,6 +812,11 @@ private static Documents.PartitionKeyDefinition GetPartitionKeyDefinition(InputP
                 return targetRanges.Single();
             }
 
+            if (isODEContinuationToken)
+            {
+                throw new InvalidOperationException("Execution of this query cannot resume using ODE Continuation token due to partition split. Please restart the query without continuation token.");
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/Query/Core/Pipeline/CosmosQueryExecutionContextFactory.cs:817]: >ODE [](http://example.com/codeflow?start=97&length=3)

spell it out

**Review by sboshra — DISMISSED** [review]
:shipit:

**Review by Maya-Painter — APPROVED** [review]

**Pull request merged**